docs/components/rule.md

```diff
@@ -1,42 +1,104 @@
+The rule component evaluates Prometheus recording and alerting rules against chosen query API via repeated `--query` (or FileSD via `--query.sd`). If more then one query is passed, round robin balancing is performed.
+ 
+Rule results are written back to disk in the Prometheus 2.0 storage format. Rule nodes at the same time participate in the system as source store nodes, which means that they expose StoreAPI and upload their generated TSDB blocks to an object store.
 
-The data of each rule node can be labeled to satisfy the clusters labeling scheme. High-availability pairs can be run in parallel and should be distinguished by the designated replica label, just like regular Prometheus servers.
+You can think of Rule as a simplified Prometheus that does not require a sidecar and does not collect and do PromQL evaluation (no QueryAPI).
+
+The data of each Rule node can be labeled to satisfy the clusters labeling scheme. High-availability pairs can be run in parallel and should be distinguished by the designated replica label, just like regular Prometheus servers.
+Read more about Ruler in HA in [here](rule.md#Ruler_HA)
 
 ```
 $ thanos rule \
-    --data-dir          "/path/to/data" \
-    --eval-interval     "30s" \
-    --rule-file         "/path/to/rules/*.rules.yaml" \
-    --alert.query-url   "http://0.0.0.0:9090" \
-    --alertmanagers.url "alert.thanos.io" \
-    --cluster.peers     "thanos-cluster.example.org" \
-    --objstore.config-file "bucket.yml"
+    --data-dir             "/path/to/data" \
+    --eval-interval        "30s" \
+    --rule-file            "/path/to/rules/*.rules.yaml" \
+    --alert.query-url      "http://0.0.0.0:9090" \ # This tells what query URL to link to in UI.
+    --alertmanagers.url    "alert.thanos.io" \
+    --query                "query.example.org" \
+    --query                "query2.example.org" \
+    --objstore.config-file "bucket.yml" \
+    --label                'monitor_cluster="cluster1"'
+    --label                'replica="A"
 ```
 
-The content of `bucket.yml`:
+## Risk
 
-```yaml
-type: GCS
-config:
-  bucket: example-bucket
-```
+Ruler has conceptual tradeoffs that might not be favorable for most use cases. The main tradeoff is its dependence on 
+query reliability. For Prometheus it is unlikely to have alert/recording rule evaluation failure as evaluation is local.
+
+For Ruler the read path is distributed, since most likely ruler is querying Thanos Querier which gets data from remote Store APIs. 
+
+This means that **query failure** are more likely to happen, that's why clear strategy on what will happen to alert and during query
+unavailability is the key.
+
+Note that Ruler will never trigger alert on it's own, so *no alert will be triggered if the query fails* without extra (mandatory)
+meta monitoring. This is exactly the same Alertmanager project. Even though there might be internal error Alertmanager will only produce metric, will 
```

Comment:
I don't understand what you are trying to say with this sentence, but maybe it's just me.

Reply:
well maybe it's too far, you are right. removed